### ISSUE TITLE
500 error for editorial queries

### ISSUE THREAD

**Issue by SydneyUni-Jim** [pkp/pkp-lib#8954]
The [_dateDiff function in pkp-lib/classes/services/queryBuilders/PKPStatsEditorialQueryBuilder.php](https://github.com/pkp/pkp-lib/blob/34052c74fadc5b74ab7958ceffa1ec70c4bbe5ec/classes/services/queryBuilders/PKPStatsEditorialQueryBuilder.php#L492) does not select the MySQL-compatible query when using the mysqlnd driver.

**Comment by asmecher** [comment]
@SydneyUni-Jim, I there are several other parts of the system that accommodate `mysql` / `mysqli` but not `mysqlnd` and would need similar adjustment:
- ADODB database layer (deprecated but still available for old upgrades, configured/instantiated in `lib/pkp/classes/install/Installer.inc.php`)
- XML-based upgrade scripts:
  - `dbscripts/xml/upgrade/3.1.1_preupdate_citations.xml`
  - `dbscripts/xml/upgrade/3.1.1_update.xml`
  - `dbscripts/xml/upgrade/3.2.0_preupdate_email_templates.xml`
  - `dbscripts/xml/upgrade/3.2.0_preupdate_versioning.xml`
  - `dbscripts/xml/upgrade/3.2.0_versioning.xml`
  - `dbscripts/xml/upgrade/event_log_oneclickuserid.xml`
- `classes/install/Upgrade.php`
- `lib/pkp/classes/install/form/InstallForm.php`

I'm curious about why you're looking to use `mysqlnd` rather than `mysqli`; it's not a need we've encountered before.

**Comment by SydneyUni-Jim** [comment]
I was under the impression that `mysqlnd` superseeds `mysqli`. I'm not a PHP person. Reading between the lines of your comment, I'm thinking I'm mistaken.

I don't have a requirement to use `mysqldnd`. It looks like I'd just been getting away with it, until now.

I'll switch to `mysqli` and close this issue and PR.

**Comment by asmecher** [comment]
Thanks for the details, @SydneyUni-Jim! I don't think mysqli is deprecated, but there is an ongoing competition between mysqli and PDO and eventually one might fully replace the other. Fortunately we don't really have to care because we're using database abstraction layers for access (first ADODB, and now increasingly Laravel) and they take care of the details, as long as we name the driver and connection details. Because we're midway through a transition from ADODB to Laravel, we have a bit of adapter code that makes sure we can access both from the same set of configuration directives, hence some of the fancy footwork to support driver names like `mysqli`, `postgres`, `postgres9`, etc. Once we are able to deprecate and remove the old pre-3.3 upgrade code we can get rid of the compatibility layer and streamline things to be consistent with modern Laravel apps.

TL;DR: You should be fine to continue with `mysqli` as the configured driver; if that changes in the future we'll document the configuration changes in the Release Notebook for that release.
